### ISSUE TITLE
Stanford parser unicodeDecodeError on phone numbers

### ISSUE THREAD

**Issue by Newtonvolt** [nltk/nltk#1211]
Python 3.4.3 windows 7
The stanford parser (3.5.2) will replace spaces with 0xa0 (which I guess is unicode non-breaking space), in the event of a phone number appearing. The stanford nltk module will not handle this well, and will crash with a UnicodeDecodeError.

To reproduce:

``` python
from nltk.parse import stanford
import os
os.environ['STANFORD_PARSER']='C:\stanford-parser\stanford-parser-full-2015-04-20'
os.environ['STANFORD_MODELS']='C:\stanford-parser\stanford-parser-full-2015-04-20'
stanford_parser = stanford.StanfordParser(model_path='C:\stanford-parser\stanford-parser-full-2015-04-20/englishPCFG.ser.gz')
x = stanford_parser.raw_parse('This is going to crash: 01 111 555.') 
```

result:

```
Traceback (most recent call last):
  File "C:/stanford-parser/crash.py", line 6, in <module>
    x = stanford_parser.raw_parse('This is going to crash: 01 111 555.')
  File "C:\Python34\lib\site-packages\nltk\parse\stanford.py", line 128, in raw_parse
    return next(self.raw_parse_sents([sentence], verbose))
  File "C:\Python34\lib\site-packages\nltk\parse\stanford.py", line 146, in raw_parse_sents
    return self._parse_trees_output(self._execute(cmd, '\n'.join(sentences), verbose))
  File "C:\Python34\lib\site-packages\nltk\parse\stanford.py", line 214, in _execute
    stdout = stdout.decode(encoding)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa0 in position 167: invalid start byte
```

**Comment by stevenbird** [comment]
Thanks @Newtonvolt

**Comment by longdt219** [comment]
Hi @Newtonvolt  and @stevenbird, 
This bug is caused by non-break char inserted between numbers. I added the code to remove this char. Tested on windows.

**Comment by alvations** [comment]
From previous experiences, there will be a lot more noise coming your way other than `0xa0`, one solution that was used in [NTU - Multilingual Corpus](http://web.mysites.ntu.edu.sg/fcbond/open/pubs/2012-ijalp-ntumc.pdf), [SeedLing Corpus](http://www.aclweb.org/anthology/W/W14/W14-2211.pdf) and [DSL Corpus Collection](https://comparable.limsi.fr/bucc2014/4.pdf) is to use a function to replace all control characters, see https://gist.github.com/alvations/55d78f627ac8bac0bf34

Also, when we clean up web-crawled, we often find the need to unescape some HTML / XML tags using https://docs.python.org/3/library/html.html. But modern web parser like `BeautifulSoup` and `Scapy` would have cleaned it up properly without the need for further escaping.

**Comment by Korred** [comment]
Hi @longdt219,
unfortunately this "fix" leads to another issue on python 2.7.10 .. please see below:

```
Traceback (most recent call last):
  File "C:\Users\SG-2\Desktop\DB-Projekt\trip_tagger\testing.py", line 27, in <module>
    x = dep_parser.raw_parse('This is going to crash: 01 111 555.')
  File "E:\Python27\lib\site-packages\nltk\parse\stanford.py", line 128, in raw_parse
    return next(self.raw_parse_sents([sentence], verbose))
  File "E:\Python27\lib\site-packages\nltk\parse\stanford.py", line 146, in raw_parse_sents
    return self._parse_trees_output(self._execute(cmd, '\n'.join(sentences), verbose))
  File "E:\Python27\lib\site-packages\nltk\parse\stanford.py", line 87, in _parse_trees_output
    res.append(iter([self._make_tree('\n'.join(cur_lines))]))
  File "E:\Python27\lib\site-packages\nltk\parse\stanford.py", line 335, in _make_tree
    return DependencyGraph(result, top_relation_label='root')
  File "E:\Python27\lib\site-packages\nltk\parse\dependencygraph.py", line 84, in __init__
    top_relation_label=top_relation_label,
  File "E:\Python27\lib\site-packages\nltk\parse\dependencygraph.py", line 328, in _parse
    assert cell_number == len(cells)
AssertionError
```

Used the example as provided by @Newtonvolt 
Issue occured after adding your two lines:

```
stdout = stdout.replace(b'\xc2\xa0',b' ')
stdout = stdout.replace(b'\xa0',b' ')
```

Edit:

Replaced `b' '` with `b'_'` just for testing... and it magically works now.. Any clue why?

**Comment by manning** [comment]
Surely the goal should be for the Python Java interface to work with Unicode, not to have to strip non-ASCII characters?!?

**Comment by alvations** [comment]
This should be resolved from #1249 =)

**Comment by alvations** [comment]
Resolved in #1249 
Closing resolved issues to prepare for v3.2.5.
